metadata-ingestion/src/datahub/ingestion/run/pipeline.py

```diff
@@ -356,74 +356,85 @@ def run(self) -> None:
         self.final_status = "unknown"
         self._notify_reporters_on_ingestion_start()
         callback = None
-        try:
-            callback = (
-                LoggingCallback()
-                if not self.config.failure_log.enabled
-                else DeadLetterQueueCallback(
-                    self.ctx, self.config.failure_log.log_config
-                )
-            )
-            for wu in itertools.islice(
-                self.source.get_workunits(),
-                self.preview_workunits if self.preview_mode else None,
+        # TODO: refactor memray code as needed.
+        if self.config.flags.generate_memory_profiles:
+            import memray
+
+            with memray.Tracker(
+                f"{self.config.flags.generate_memory_profiles}/{self.config.run_id}.bin"
             ):
```

Comment:
don't think this works - if `generate_memory_profiles` is not set, then we don't do anything

btw you can use contextlib.ExitStack or NullContext to conditionally enter a `with` clause

Reply:
This was an oversight on my part. Already fixed.